src/controller/CommissioningDelegate.h

```diff
@@ -427,6 +427,21 @@ class CommissioningParameters
         return *this;
     }
 
+    // Whether network setup should be performed.  If not set, the need for
+    // network setup will be auto-detected based on whether we established a
+    // PASE session on-network or not.
+    Optional<bool> GetDoNetworkSetup() const { return mDoNetworkSetup; }
```

Comment:
I think this might be easier to understand at first glance if you use an enum similar to what you have below and default to auto.

Reply:
Yes, I can do that.

Reply:
@cecille Please take a look?